src/main/java/frc/trigon/robot/subsystems/arm/Arm.java

```diff
@@ -0,0 +1,212 @@
+package frc.trigon.robot.subsystems.arm;
+
+import com.ctre.phoenix.motorcontrol.can.TalonSRX;
+import com.ctre.phoenix6.hardware.CANcoder;
+import com.revrobotics.CANSparkMax;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.trajectory.TrapezoidProfile;
+import edu.wpi.first.math.util.Units;
+import edu.wpi.first.wpilibj.Timer;
+import edu.wpi.first.wpilibj2.command.*;
+import frc.trigon.robot.utilities.Conversions;
+
+import java.util.Set;
+
+public class Arm extends SubsystemBase {
+    private final static Arm INSTANCE = new Arm();
+    private final CANSparkMax
+            masterAngleMotor = ArmConstants.MASTER_ANGLE_MOTOR,
+            followerAngleMotor = ArmConstants.FOLLOWER_ANGLE_MOTOR,
+            masterElevatorMotor = ArmConstants.MASTER_ELEVATOR_MOTOR,
+            followerElevatorMotor = ArmConstants.FOLLOWER_ELEVATOR_MOTOR;
+    private final CANcoder angleEncoder = ArmConstants.ANGLE_ENCODER;
+    private final TalonSRX elevatorEncoder = ArmConstants.ELEVATOR_ENCODER;
+    private TrapezoidProfile
+            angleMotorProfile = null,
+            elevatorMotorProfile = null;
+    private double
+            lastAngleMotorProfileGenerationTime,
+            lastElevatorMotorProfileGenerationTime;
+
+    public static Arm getInstance() {
+        return INSTANCE;
+    }
+
+    private Arm() {
+
+    }
+
+    public Command getSetTargetArmPositionCommand(Rotation2d targetAngle, double targetElevatorPosition) {
+        return getSetTargetArmPositionCommand(targetAngle, targetElevatorPosition, 100, 100);
+    }
+
+    /**
+     * sets arm to target state
+     * @param targetState             target state of the arm
+     * @param angleSpeedPercentage    target speed for the angle motor
+     * @param elevatorSpeedPercentage target speed for the elevator motor
+     * @return a command that sets the arm to targetState at the speed set by angleSpeedPercentage and elevatorSpeedPercentage
+     */
+    public Command getSetTargetArmStateCommand(ArmConstants.ArmState targetState, double angleSpeedPercentage, double elevatorSpeedPercentage) {
+        return getSetTargetArmPositionCommand(targetState.angle, targetState.elevatorPosition, angleSpeedPercentage, elevatorSpeedPercentage);
+    }
+
+    public Command getSetTargetArmStateCommand(ArmConstants.ArmState targetState) {
+        return getSetTargetArmPositionCommand(targetState.angle, targetState.elevatorPosition, 100, 100);
+    }
+
+    /**
+     * sets arm to target state
```

Comment:
here you need a capitalized letter and a period at the end of a sentence.
Say something like "Creates a command that ..."